### PR TITLE
[#277] dedicated for is saved in db

### LINKED ISSUE #277: Change field dedicated_for to DB table
Actually, field dedicated_for is changed by me to an array of strings.
But in fact, it's an enum and the values of it are in https://docs.google.com/spreadsheets/d/1tQAHG-Hk8IhmSXQaR5BULiZobx1UNtfLUpK9CMqMeuE/edit?ts=5bd0b4c6#gid=1731955460
in the tab, "service description content" in cells B5:B10
For one service it should be possible to choose multiple receivers of the service.

## Solution
New DB table and many to many relation should be created connected to the `Service` Active record.
Additionally changes following things need to be updated:

* `db/data.yml` and `db/seeds.rb`
* `searchable.rb` concern - filtering dedicated for should take into account new schema

**EDIT**: (@michal-szostak) - I changed ticket description to match our current conclusion.

### PR DESCRIPTION
**Pull request by martaswiatkowska**
Fix #277

### CODE REVIEW

File: app/controllers/concerns/service/searchable.rb [hunk 1]
```diff
@@ -66,26 +67,21 @@ def indented_name(name, level)
...
-      end
-      [field[0], field[1], query.count]
-    end.select do |element|
-      element[2] != 0
+  def target_groups_options(category = nil)
+    query = TargetGroup.select("target_groups.name, target_groups.id, count(service_target_groups.service_id) as service_count")
+                .joins(:categories)
+
+    unless category.nil?
+      query = query.where("categories.id = ?", category.id)
```
Comment (mkasztelnik) [inline app/controllers/concerns/service/searchable.rb]: You can use short version:

```ruby
query = query.where("categories.id = ?", category.id) if category
```

or even:

```ruby
query = query.where(categories: category) if category
```

File: app/helpers/service_helper.rb [hunk 2]
```diff
@@ -32,7 +32,7 @@ def get_terms_and_condition_hint_text(service)
   end
 
   def dedicated_for_links(service)
-    service.dedicated_for.map { |target| link_to(target, services_path(dedicated_for: target)) }
+    service.target_groups.map { |target| link_to(target.name, services_path(target_groups: target)) }
```
Comment (kmarszalek) [inline app/helpers/service_helper.rb]: after rebase to master you will have second method in this helper that uses dedicated_for

File: app/controllers/concerns/service/searchable.rb (hunk 1 above)
Comment (martaswiatkowska) [reply app/controllers/concerns/service/searchable.rb]: I think not

Comment (mkasztelnik) [reply app/controllers/concerns/service/searchable.rb]: Yeap you can, I checked it.

**Pull request merged**